modules/twinklewarn.js

```diff
@@ -1202,31 +1203,34 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 	// Trigger custom label/change on main category change
 	Twinkle.warn.callback.change_subcategory(e);
 
-	// Build select menu with jquery.chosen
-	if (!Twinkle.getPref('oldSelect')) {
+	// Use select2 to make the select menu searchable
+	if (!Twinkle.getPref('oldSelect') && $.fn.select2) {
 		$('select[name=sub_group]')
-			.chosen({width: '100%', search_contains: true})
+			.select2({ width: '100%' })
 			.change(Twinkle.warn.callback.change_subcategory);
 
+		// if main_group is level1/2/3/4/4im, the ul has option groups,
+		if ($('select[name=main_group]').val().indexOf('level') === 0) {
+			Morebits.select2SearchHighlights($('select[name=sub_group]'), true);
+		} else {
+			Morebits.select2SearchHighlights($('select[name=sub_group]'), false);
+		}
+
 		mw.util.addCSS(
-			// Force chosen select menu to display over the dialog while overflowing
-			// based on https://github.com/harvesthq/chosen/issues/1390#issuecomment-21397245
-			'.ui-dialog.morebits-dialog .morebits-dialog-content { overflow:visible !important; }' +
-			'.ui-dialog.morebits-dialog { overflow: inherit !important; }' +
+			// prevent dropdown from appearing behind the dialog
+			'.select2-container { z-index: 10000; }' +
```

Comment:
Forgot to mention this, but was this an issue for you or were you just being cautious?  I didn't see any issues without it in chrome.

Reply:
The dropdown did disappear behind the dialog at least on one instance. Can't replicate that now, but I guess this is good just to be safe?

Reply:
We can revisit later if need be, but maybe just comment to that effect (‘just in case‘) for now, no need to spend time testing right now